src/warnings.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "warnings.h"
+
+#include "sync.h"
+#include "clientversion.h"
+#include "util.h"
+
+RecursiveMutex cs_warnings;
+std::string strMiscWarning;
+bool fLargeWorkForkFound = false;
+bool fLargeWorkInvalidChainFound = false;
+
+void SetMiscWarning(const std::string& strWarning)
+{
+    LOCK(cs_warnings);
+    strMiscWarning = strWarning;
+}
+
+void SetfLargeWorkForkFound(bool flag)
+{
+    LOCK(cs_warnings);
+    fLargeWorkForkFound = flag;
+}
+
+bool GetfLargeWorkForkFound()
+{
+    LOCK(cs_warnings);
+    return fLargeWorkForkFound;
+}
+
+void SetfLargeWorkInvalidChainFound(bool flag)
+{
+    LOCK(cs_warnings);
+    fLargeWorkInvalidChainFound = flag;
+}
+
+bool GetfLargeWorkInvalidChainFound()
+{
+    LOCK(cs_warnings);
+    return fLargeWorkInvalidChainFound;
+}
+
+std::string GetWarnings(const std::string& strFor)
+{
+    std::string strStatusBar;
+    std::string strRPC;
+    std::string strGUI;
+    const std::string uiAlertSeperator = "\n";
+
+    LOCK(cs_warnings);
+
+    if (!CLIENT_VERSION_IS_RELEASE) {
+        strStatusBar = "This is a pre-release test build - use at your own risk - do not use for staking or merchant applications";
+        strGUI = _(strStatusBar.c_str());
```

Comment:
need to use the `_()` method on the string itself and not the variable, otherwise it never makes it to transifex for translation. Same with the other strings below.

Also, to avoid invalidation of an already translated string on transifex, need to ad a `!` at the end of the string (ie `do not use for staking or merchant applications!`)

Reply:
k done, reverted it to the previous state.